src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -239,10 +246,89 @@ ProjectItemInstance projectReference
                         $"{setConfigurationString};{setPlatformString};{setTargetFrameworkString}").ToImmutableDictionary();
                 }
             }
-
             return new GlobalPropertyPartsForMSBuildTask(newProperties, defaultParts.AdditionalProperties, newUndefineProperties);
         }
 
+        private static string GetCompatiblePlatform(ProjectItemInstance projectReference){
+                
+                var ReferencedProject = new Project(projectReference.EvaluatedInclude);
+                string projectReferencePlatformMetadata =  ReferencedProject.GetPropertyValue(PlatformMetadataName);
+                var CurrentProjectPlatform = projectReference.Project.GetPropertyValue(PlatformMetadataName);
+                var platformLookupTable = projectReference.Project.GetPropertyValue(PlatformLookupTableMetadataName);
+                var referencedLookupTableMetadata = ReferencedProject.GetPropertyValue(PlatformLookupTableMetadataName);
+
+#nullable enable
+            Dictionary<string, string>? currentProjectLookupTable = ExtractLookupTable(platformLookupTable);
+#nullable disable          
+#nullable enable
+            Dictionary<string, string>? projectReferenceLookupTable = ExtractLookupTable(referencedLookupTableMetadata);
+#nullable disable
+            HashSet<string> projectReferencePlatforms = new HashSet<string>();
+
+
+                foreach (string s in projectReferencePlatformMetadata.Split(MSBuildConstants.SemicolonChar, StringSplitOptions.RemoveEmptyEntries))
```

Comment:
This is copied from the `GetCompatiblePlatform` task. This code should be shared instead of duplicated to avoid divergence.

Reply:
Moved to a shared location